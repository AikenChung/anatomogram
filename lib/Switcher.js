'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _urijs = require('urijs');

var _urijs2 = _interopRequireDefault(_urijs);

require('./Switcher.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var resolveUrlToIcon = function resolveUrlToIcon(urlToResources, selectedType, anatomogramType) {
  return (0, _urijs2.default)('img/' + (selectedType === anatomogramType ? '' : 'un') + 'selected' + anatomogramType + '.png', urlToResources).toString();
};

var Switcher = function Switcher(props) {
  return _react2.default.createElement(
    'div',
    { style: { float: 'left' } },
    props.anatomogramTypes.map(function (anatomogramType) {
      return _react2.default.createElement('img', { key: anatomogramType,
        className: 'gxa-selection-icon',
        onClick: function onClick() {
          props.onClick(anatomogramType);
        },
        src: resolveUrlToIcon(props.urlToResources, props.selectedType, anatomogramType) });
    })
  );
};

Switcher.propTypes = {
  anatomogramTypes: _propTypes2.default.arrayOf(_propTypes2.default.string).isRequired,
  selectedType: _propTypes2.default.string.isRequired,
  onClick: _propTypes2.default.func.isRequired,
  urlToResources: _propTypes2.default.string.isRequired
};

exports.default = Switcher;