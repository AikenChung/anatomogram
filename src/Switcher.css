.gxa-selection-icon {
  border: 1px solid #ccc;
  border-radius: 4px;
  width: 100%;
  height: auto;
  padding: 2px;
}
.gxa-selection-icon:hover {
  border: 1px solid orange;
  background: lightgoldenrodyellow;
  cursor: pointer;
}
.rounded-corners {
  border: 1px solid #ccc;
  border-radius: 4px;
}
.right-dimensions {
  width: 100%;
  height: auto;
  padding: 2px;
}
