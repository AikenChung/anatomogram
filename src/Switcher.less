.gxa-selection-icon {
  .rounded-corners;
  .right-dimensions;
  &:hover {
    border: 1px solid orange;
    background: lightgoldenrodyellow;
    cursor: pointer;
  }
}

.rounded-corners {
  border: 1px solid #ccc;
  border-radius: 4px;
}

.right-dimensions {
  width: 100%;
  height: auto;
  padding: 2px;
}
