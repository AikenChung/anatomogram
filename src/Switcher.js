import React from 'react'
import PropTypes from 'prop-types'

import URI from 'urijs'

import './Switcher.css'

const resolveUrlToIcon = (urlToResources, selectedType, anatomogramType, ) =>
  URI(`img/${selectedType === anatomogramType ? `` : `un`}selected${anatomogramType}.png`, urlToResources).toString()

const Switcher = (props) =>
  <div style={{display: `inline-block`, verticalAlign: `top`, width: `7%`}}>
    {props.anatomogramTypes.map((anatomogramType) =>
      <img key={anatomogramType}
           className={`gxa-selection-icon`}
           onClick={() => {props.onClick(anatomogramType)}}
           src={resolveUrlToIcon(props.urlToResources, props.selectedType, anatomogramType)} />
    )}
  </div>

Switcher.propTypes = {
  anatomogramTypes: PropTypes.arrayOf(PropTypes.string).isRequired,
  selectedType: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  urlToResources: PropTypes.string.isRequired
}

export default Switcher
