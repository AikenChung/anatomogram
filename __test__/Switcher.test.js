import React from 'react'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'
import sinon from 'sinon'

import Switcher from '../src/Switcher'
import availableSpecies from '../src/svgs.json'

describe(`Anatomogram switcher`, () => {
  // it(`should render without throwing an error`, () => {
  //   expect(shallow(<Switcher />).contains(<div className="foo">Bar</div>)).toBe(true)
  // })
  const requiredProps = {
    anatomogramTypes: [`female`, `male`],
    selectedType: `female`,
    onClick: () => {},
    urlToResources: ``
  }

  test(`should contain as many buttons as anatomogram types`, function() {
    expect(shallow(<Switcher {...requiredProps} anatomogramTypes={[]} selectedType={``}/>).find(`img`)).toHaveLength(0)
    expect(shallow(<Switcher {...requiredProps} />).find(`img`)).toHaveLength(2)
  })

  test(`should contain img selectable by class "gxa-selection-icon"`, () => {
    expect(shallow(<Switcher {...requiredProps} anatomogramTypes={[]} selectedType={``}/>).find(`.gxa-selection-icon`)).toHaveLength(0)
    expect(shallow(<Switcher {...requiredProps} />).find(`.gxa-selection-icon`)).toHaveLength(2)
  })

  test(`should respond to onClick events with the anatomogram type`, () => {
    const onButtonClick = sinon.spy()
    const wrapper = shallow(<Switcher {...requiredProps} onClick={onButtonClick}/>)
    wrapper.find(`img`).first().simulate(`click`);
    wrapper.find(`img`).last().simulate(`click`);
    expect(onButtonClick.calledTwice).toBe(true);
    expect(onButtonClick.calledWith(`female`)).toBe(true);
    expect(onButtonClick.calledWith(`male`)).toBe(true);
  })

  Object.keys(availableSpecies).forEach((species) => {
    test(`matches snapshot for ${species}`, () => {
      const anatomogramTypes = availableSpecies[species]
      const tree = renderer.create(<Switcher {...requiredProps} anatomogramTypes={anatomogramTypes} selectedType={anatomogramTypes[0] || ``}/>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
