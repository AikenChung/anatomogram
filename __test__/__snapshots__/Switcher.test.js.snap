// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anatomogram switcher matches snapshot for anolis_carolinensis 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for arabidopsis_thaliana 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for bos_taurus 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for brachypodium_distachyon 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_whole_plant.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_flower_parts.png"
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for gallus_gallus 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for homo_sapiens 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_brain.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_female.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_male.png"
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for hordeum_vulgare 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_whole_plant.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_flower_parts.png"
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for macaca_mulatta 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for monodelphis_domestica 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for mus_musculus 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_brain.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_female.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_male.png"
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for oryza_sativa 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_whole_plant.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_flower_parts.png"
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for papio_anubis 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for rattus_norvegicus 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for solanum_lycopersicum 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_whole_plant.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_flower_parts.png"
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for solanum_tuberosum 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for sorghum_bicolor 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_whole_plant.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_flower_parts.png"
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for tetraodon_nigroviridis 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for triticum_aestivum 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_whole_plant.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_flower_parts.png"
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for xenopus_tropicalis 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
/>
`;

exports[`Anatomogram switcher matches snapshot for zea_mays 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "maxWidth": "32px",
      "verticalAlign": "top",
      "width": "10%",
    }
  }
>
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/selected_whole_plant.png"
  />
  <img
    className="gxa-selection-icon"
    onClick={[Function]}
    src="img/unselected_flower_parts.png"
  />
</div>
`;
